kotlin/codegen/src/test/java/com/squareup/moshi/kotlin/codegen/JsonClassCodegenProcessorTest.kt

```diff
@@ -15,367 +15,338 @@
  */
 package com.squareup.moshi.kotlin.codegen
 
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile
+import com.tschuchort.compiletesting.SourceFile.Companion.kotlin
 import org.assertj.core.api.Assertions.assertThat
-import org.jetbrains.kotlin.cli.common.ExitCode
 import org.junit.Ignore
 import org.junit.Rule
 import org.junit.Test
 import org.junit.rules.TemporaryFolder
-import javax.annotation.processing.Processor
 
 /** Execute kotlinc to confirm that either files are generated or errors are printed. */
 class JsonClassCodegenProcessorTest {
   @Rule @JvmField var temporaryFolder: TemporaryFolder = TemporaryFolder()
 
   @Test fun privateConstructor() {
-    val call = KotlinCompilerCall(temporaryFolder.root)
-    call.inheritClasspath = true
-    call.addService(Processor::class, JsonClassCodegenProcessor::class)
-    call.addKt("source.kt", """
-        |import com.squareup.moshi.JsonClass
-        |
-        |@JsonClass(generateAdapter = true)
-        |class PrivateConstructor private constructor(var a: Int, var b: Int) {
-        |  fun a() = a
-        |  fun b() = b
-        |  companion object {
-        |    fun newInstance(a: Int, b: Int) = PrivateConstructor(a, b)
-        |  }
-        |}
-        |""".trimMargin())
-
-    val result = call.execute()
-    assertThat(result.exitCode).isEqualTo(ExitCode.COMPILATION_ERROR)
-    assertThat(result.systemErr).contains("constructor is not internal or public")
+    val result = compile(kotlin("source.kt",
+        """
+          import com.squareup.moshi.JsonClass
+          
+          @JsonClass(generateAdapter = true)
+          class PrivateConstructor private constructor(var a: Int, var b: Int) {
+            fun a() = a
+            fun b() = b
+            companion object {
+              fun newInstance(a: Int, b: Int) = PrivateConstructor(a, b)
+            }
+          }
+          """
```

Comment:
why the change from a margin and trimMargin? I think I prefer the old string

Reply:
trimMargin breaks syntax highlighting, [which the `kotlin()` factory enables](https://github.com/tschuchortdev/kotlin-compile-testing/pull/13).

![image](https://user-images.githubusercontent.com/1361086/65637040-7e23ed80-dfb1-11e9-9997-32e117c91daa.png)

![image](https://user-images.githubusercontent.com/1361086/65637080-8f6cfa00-dfb1-11e9-8160-5b9f3008ae61.png)

![image](https://user-images.githubusercontent.com/1361086/65637108-9c89e900-dfb1-11e9-9262-51a18958670f.png)